bin/dynamic_config/set_hoc_flags.rb

```diff
@@ -0,0 +1,83 @@
+#!/usr/bin/env ruby
+#
+# A script to set the Gatekeeper and DCDO flags for HOC 2015 in one of
+# several predefined modes:
+#
+# - red ("read-only" disaster mode for HOC scripts)
+# - yellow (non-essential features disabled for HOC scripts)
+# - green (all features enabled)
+#
+# See the documentation below for more details.
+puts "Loading..."
+require_relative '../../dashboard/config/environment'
+require 'pp'
+
+# The names of scripts we expect to see
+HIGH_SCALE_SCRIPT_NAMES = %w(frozen gumball hourofcode starwars starwarsblocks mc)
+
+def main(argv)
+  if argv.size != 1
+
+    puts <<-EOF
+Usage: #{$0}: red|yellow|green
+
+ - red: No HOC sharing, no HOC hints, no hoc_activity writes, no signed-in user progress saving, no HOC puzzle rating.
+ - yellow: No HOC puzzle rating, no HOC hints, no hoc_activity writes.
+ - green: all features turned on. SHOULD NOT BE USED UNTIL HOC IS OVER
+
+ Public caching is enabled for all HOC scripts in all cases because this doesn't take away functionality
+EOF
+    exit
+  end
+
+  mode = argv.shift
+  # Set global Gatekeeper and DCDO flags.
+  case mode
+    when 'red'
+      Gatekeeper.set('postMilestone', value: false)
+      Gatekeeper.set('hoc_activity_writes_disabled', value: true)
+      Gatekeeper.set('async_activity_writes', value: true)
+      Gatekeeper.set('tracking_pixel_enabled', value: false)
+      Gatekeeper.set('puzzle_rating', value: false)
+      DCDO.set('activity_max_rate', 100)
+    when 'yellow'
+      Gatekeeper.set('postMilestone', value: true)
+      Gatekeeper.set('hoc_activity_writes_disabled', value: true)
+      Gatekeeper.set('tracking_pixel_enabled', value: false)
+      Gatekeeper.set('async_activity_writes', value: true)
+      Gatekeeper.set('puzzle_rating', value: false)
+      DCDO.set('activity_max_rate', 3000)  # Max async op rate per queue processor.
+    when 'green'
+      Gatekeeper.set('postMilestone', value: true)
+      Gatekeeper.set('hoc_activity_writes_disabled', value: false)
+      Gatekeeper.set('async_activity_writes', value: false)
```

Comment:
That flag only affects whether writes are enqueued; processing will continue unabated.

The reason to switch it off is just cost (so that every write doesn't have to go through the queue) and to permit us to migrate away from async writes if we decide we want to to go some other path (e.g. migration to DynamoDB or sarhing.)